scripts/redis/gpt5_cache_migration.py

```
@@ -0,0 +1,385 @@
+
+class KeyTransformer:
+    """Handles transformation of cache key patterns"""
+
+    # Key transformation patterns
+    PATTERNS = [
+        (r"gpt-4\.1:cache:", "gpt5:cache:"),
+        (r"gpt41:cache:", "gpt5:cache:"),
+        (r"gpt-4-turbo:cache:", "gpt5:cache:"),
+        (r"cache:gpt-4:", "gpt5:cache:"),
+        (r"gpt4:cache:", "gpt5:cache:"),
+    ]
```

Ruff (0.14.4)

Mutable class attributes should be annotated with `typing.ClassVar`

(RUF012)